test/e2e/framework/util.go

```diff
@@ -2757,47 +2757,117 @@ func WaitForPodsWithLabel(c clientset.Interface, ns string, label labels.Selecto
 	return
 }
 
-// DeleteRCAndPods a Replication Controller and all pods it spawned
-func DeleteRCAndPods(clientset clientset.Interface, internalClientset internalclientset.Interface, ns, name string) error {
-	By(fmt.Sprintf("deleting replication controller %s in namespace %s", name, ns))
-	rc, err := clientset.Core().ReplicationControllers(ns).Get(name)
+func getRuntimeObjectForKind(c clientset.Interface, kind schema.GroupKind, ns, name string) (runtime.Object, error) {
+	switch kind {
+	case api.Kind("ReplicationController"):
+		return c.Core().ReplicationControllers(ns).Get(name)
+	case extensionsinternal.Kind("ReplicaSet"):
+		return c.Extensions().ReplicaSets(ns).Get(name)
+	case extensionsinternal.Kind("Deployment"):
+		return c.Extensions().Deployments(ns).Get(name)
+	default:
+		return nil, fmt.Errorf("Unsupported kind: %v", kind)
+	}
+}
+
+func deleteResource(c clientset.Interface, kind schema.GroupKind, ns, name string, deleteOption *v1.DeleteOptions) error {
+	switch kind {
+	case api.Kind("ReplicationController"):
+		return c.Core().ReplicationControllers(ns).Delete(name, deleteOption)
+	case extensionsinternal.Kind("ReplicaSet"):
+		return c.Extensions().ReplicaSets(ns).Delete(name, deleteOption)
+	case extensionsinternal.Kind("Deployment"):
+		return c.Extensions().Deployments(ns).Delete(name, deleteOption)
+	default:
+		return fmt.Errorf("Unsupported kind: %v", kind)
+	}
+}
+
+func getSelectorFromRuntimeObject(obj runtime.Object) (*unversioned.LabelSelector, error) {
```

Comment:
We should return "labels.Selector" from this method (as this is the generic interface). The further code will be simpler then.